tests/test_config/test_config.py

```diff
@@ -292,6 +293,24 @@ def check(
     assert counters[type(x['b'])] == 2
 
 
+def test_uncacheable_config_warning(make_app, tmp_path):
```

Comment:
No need for a full application. You can just construct an configuration object and check pickability yourself and patch `sphinx.config.logger` as in the other tests.

Reply:
my only concern here would be then this would be a more "indirect" test of what the PR is aiming to fix, i.e. that a warning is emitted in the `sphinx-build`,
and I'm also not a big fan of mocking / patching in tests, as again it makes the test more "indirect" and more prone to not testing what it is actually meant to be testing

Is it a sticking point if I don't change this test?

Reply:
Personally, I would have kept the logic as isolated as possible since you are in `test_config`. If you to keep the test that way, I'd suggest instead putting in another test module (probably in `test_builders`, possibly using the dummy application instead).

If you want to keep it in `test_config`, I would prefer to use a standalone `Config()` object. Technically speaking, if the warning is not emitted during the build, then this should be tested separately (but I think we are already testing this logic). And not through this (because if the warning is not emitted then the issue is not related to your PR but rather to the logger directly).

Reply:
Okie, maybe I move it out of `test_config` then, 
because yeh I would kind of like to keep this as a "test-driven development" type test, whereby I'm testing exactly for the issue I'm actually trying to address in the PR

Reply:
So you want to make it more E2E rather than checking `__getstate__` itself.

Reply:
yeh I guess so